packages/data-service/src/data-service.ts

```diff
@@ -2357,19 +2372,36 @@ class DataServiceImpl extends WithLogContext implements DataService {
             .sort({ _id: 1 })
             .toArray();
 
+          if (session.inTransaction()) {
+            await session.abortTransaction();
+            await session.endSession();
+          }
+
           const changes = docsToPreview.map((before, idx) => ({
             before,
             after: changedDocs[idx],
           }));
           return { changes };
-        } finally {
-          await session.abortTransaction();
-          await session.endSession();
+        } catch (err: any) {
+          if (isTransactionAbortError(err)) {
```

Comment:
We _might_ be able to avoid this if we check if the transaction has been aborted or the session otherwise closed after every async call above, but I don't think we can really avoid that. The session can be aborted/killed at any point if we pass in an abortSignal.